src/runtime/composables/useModal.ts

```diff
@@ -0,0 +1,74 @@
+import { ref, inject } from 'vue'
+import type { ShallowRef, Component, InjectionKey } from 'vue'
+import { createSharedComposable } from '@vueuse/core'
+import { mergeConfig } from '../utils'
+import type { ComponentProps, Strategy } from '../types'
+// @ts-expect-error
+import appConfig from '#build/app.config'
+import { modal } from '#ui/ui.config'
+
+export interface ModalState {
+  component: Component | string
+  props: ModalProps
+}
+
+export const modalInjectionKey: InjectionKey<ShallowRef<ModalState>> = Symbol('nuxt-ui.modal')
+
+const config = mergeConfig<typeof modal>(appConfig.ui.strategy, appConfig.ui.modal, modal)
+
+interface ModalProps {
+  appear?: boolean,
+  overlay?: boolean,
+  transition?: boolean,
+  preventClose?: boolean,
+  fullscreen?: boolean,
+  class?: string | Object | string[]
+  ui?: Partial<typeof config> & { strategy?: Strategy }
+  onClose?: () => void
+  onClosePrevented?: () => void
+  'onUpdate:modelValue'?: (value: boolean) => void
+}
+
+function _useModal () {
+  const modalState = inject(modalInjectionKey)
+  
+  const isOpen = ref(false)
+
+  function reveal<T extends Component> (component: T, props?: ModalProps & ComponentProps<T>) {
```

Comment:
Can we call this `open` instead of `reveal`?

Reply:
Renamed in 9e0a47cccda1a4a9e85fd5d3da42cb2c2d5809d2